coordinator/src/orderbook/trading.rs

```diff
@@ -0,0 +1,223 @@
+use crate::orderbook::db;
+use crate::orderbook::routes::MatchParams;
+use crate::orderbook::routes::Order;
+use crate::orderbook::routes::OrderType;
+use anyhow::Result;
+use diesel::r2d2::ConnectionManager;
+use diesel::r2d2::PooledConnection;
+use diesel::PgConnection;
+use rust_decimal::Decimal;
+use std::cmp::Ordering;
+use trade::Direction;
+
+/// Matches a provided market order with limit orders from the DB
+///
+/// If the order is a long order, we return the orders with the highest price
+/// If the order is a short order, we return the orders with the lowest price
+pub fn match_order(
+    order: Order,
+    db_connection: &mut PooledConnection<ConnectionManager<PgConnection>>,
```

Comment:
🔧 It would be better if this was written as a pure function without involving the database. Loading the orders from the DB should happen outside. Then you can write unit tests.

Reply:
Good idea :)

Reply:
[`94d65ee` (#248)](https://github.com/get10101/10101/pull/248/commits/94d65eed8af90a0ccd3fa10a24614f6f9f487e06)